sdcm/nemesis.py

```diff
@@ -3537,7 +3538,19 @@ def disrupt_memory_stress(self):
         Try to run stress-ng to preempt allocated memory of scylla process,
         we don't monitor swap usage in /proc/$scylla_pid/status, just make sure
         no coredump, serious db error occur during the heavy load of memory.
+
+        For k8s cluster uses chaos-mesh experiment based on https://github.com/chaos-mesh/memStress
         """
+        if self._is_it_on_kubernetes():
+            self.log.info('Try to allocate 120% available memory, the allocated memory will be swaped out')
+            experiment = MemoryStressExperiment(pod=self.target_node, duration="100s", workers=12, size="10%")
```

Comment:
Why are we using 12 workers here ? with stress-ng we didn't use multiple processes/threads, why do we need it here ? does it make any difference ?

Also why on each command we are using a different number of workers ?

Reply:
I just followed some examples. But apparently it is not working correctly:
![image](https://user-images.githubusercontent.com/18242288/210065368-880cb62a-c1b0-495d-a4e8-d8073a36fd54.png)

I'll try with different settings

Reply:
Using approach like with typical nodes (getting available memory from `/proc/meminfo`) is not working correctly - all the memory is not occupied:
![image](https://user-images.githubusercontent.com/18242288/210088576-37b6d062-033f-4ea5-99e1-5c413ec8b3a6.png)

My guess is that scylla container gets additional memory as needed, so getting `/proc/meminfo` value before scylla occupies all available memory leads to this error..
I'll try with percentage option for this experiment (but it accepts only up to 100%, so 120% case won't be covered) and update here.

Reply:
with 100% limit got similar result, but I found we have something like `k8s_cluster.calculated_memory_limit` which should be right - testing it.

Reply:
result with using `1.2 * k8s_cluster.calculated_memory_limit`
![image](https://user-images.githubusercontent.com/18242288/210107912-95f556f9-4845-4b08-bd52-52dddd21c006.png)
Still not all the memory is occupied - I will try few more ways.

Reply:
I tried with memory stress ramp up over 30s, similar result:
![image](https://user-images.githubusercontent.com/18242288/210240594-db4c8211-f3cc-45eb-9764-20ddd96836b5.png)

Reply:
Ok, I think I've found a reason for this behavior: I've read that k8s memory management kills a pod when memory usage raises to quickly. This relates to that 90% mem stress shows higher mem usage as it raises slower than for 120% stress.
When I prolonged stress ramp up to 60s, result looks like this:
![image](https://user-images.githubusercontent.com/18242288/210255960-2759ca82-60b9-4a46-a87b-900b47685e7d.png)

Still not perfect, but it's getting there.

Reply:
resutlts when getting 1gb/s ramp up:
![image](https://user-images.githubusercontent.com/18242288/210316676-c9dde651-cfb3-4fae-b2d1-5dd18ad5edbc.png)
Still I can see oomKill's in logs.  What I saw from VM based test results, this nemesis kills the scylla too. So, do we care? Can I leave it as it is?

Reply:
oomKill that kills scylla process/pod ? that's good, i.e. the system behaves as expected, we should be aware that we'll need to add code to wait for it to recover, before moving on.

in VM we have a few issues that if we do that long enough, scylla doesn't crash but gets unresponsive, and main c-s load fails.  in some cases we only have reactor stalls.
I don't think we got OOM killed

Reply:
maybe we should switch order of memory stress size: first make it 90% and then go to 100% wouldn't it be more natural?